packages/@react-spectrum/actionbar/src/ActionBar.tsx

```diff
@@ -82,6 +82,67 @@ const ActionBarInner = React.forwardRef((props: ActionBarInnerProps, ref: DOMRef
     announce(formatMessage('actionsAvailable'));
   }, [formatMessage]);
 
+  let restoreFocusRef = useRef<HTMLElement>(null);
+  let containerRef = useRef<HTMLElement>(null);
+
+  useEffect(() => {
+    // Event handler to keep track of focus changes within ActionBarContainer when ActionBar is open.
+    let timeoutId = null;
+    let onFocusContainer = (e:FocusEvent) => {
+      let activeElement = e.target as HTMLElement;
+      if (timeoutId) {
+        cancelAnimationFrame(timeoutId);
+        timeoutId = null;
+      }
+      timeoutId = requestAnimationFrame(() => {
+        if (domRef.current && !domRef.current.contains(activeElement)) {
+          restoreFocusRef.current = activeElement;
+        }
+      });
+    };
```

Comment:
Still seeing issues with where focus is being restored.
1. Using the keyboard select row 1
2. Move focus to "Foo 3"
3. Tab to ActionBar
4. Scroll to bottom of TableView via scroll wheel/touch pad
5. Hit Escape
Focus moves to row 13 or some other rows. Did https://github.com/adobe/react-spectrum/pull/2451#discussion_r750694193 not work when you tried it?

Reply:
Interesting. Looks like adding the cancelAnimationFrame works for [Storybook-17](https://reactspectrum.blob.core.windows.net/reactspectrum/72a465e4cf25ce92deb638bacbe7323497e11757/storybook-17/index.html?path=/story/actionbar--default), but breaks [Storybook](https://reactspectrum.blob.core.windows.net/reactspectrum/72a465e4cf25ce92deb638bacbe7323497e11757/storybook/index.html?path=/story/actionbar--default) now.